Sources/TensorFlow/Core/Utilities.swift

```diff
@@ -167,3 +167,176 @@ extension UnsafeMutablePointer where Pointee == CTensorHandle? {
+            0x2de92c6f592b0275, 0x4a7484aa6ea6e483, 0x5cb0a9dcbd41fbd4, 0x76f988da831153b5,
+            0x983e5152ee66dfab, 0xa831c66d2db43210, 0xb00327c898fb213f, 0xbf597fc7beef0ee4,
+            0xc6e00bf33da88fc2, 0xd5a79147930aa725, 0x06ca6351e003826f, 0x142929670a0e6e70,
+            0x27b70a8546d22ffc, 0x2e1b21385c26c926, 0x4d2c6dfc5ac42aed, 0x53380d139d95b3df,
+            0x650a73548baf63de, 0x766a0abb3c77b2a8, 0x81c2c92e47edaee6, 0x92722c851482353b,
+            0xa2bfe8a14cf10364, 0xa81a664bbc423001, 0xc24b8b70d0f89791, 0xc76c51a30654be30,
+            0xd192e819d6ef5218, 0xd69906245565a910, 0xf40e35855771202a, 0x106aa07032bbd1b8,
+            0x19a4c116b8d2d0c8, 0x1e376c085141ab53, 0x2748774cdf8eeb99, 0x34b0bcb5e19b48a8,
+            0x391c0cb3c5c95a63, 0x4ed8aa4ae3418acb, 0x5b9cca4f7763e373, 0x682e6ff3d6b2b8a3,
+            0x748f82ee5defb2fc, 0x78a5636f43172f60, 0x84c87814a1f0ab72, 0x8cc702081a6439ec,
+            0x90befffa23631e28, 0xa4506cebde82bde9, 0xbef9a3f7b2c67915, 0xc67178f2e372532b,
+            0xca273eceea26619c, 0xd186b8c721c0c207, 0xeada7dd6cde0eb1e, 0xf57d4f7fee6ed178,
+            0x06f067aa72176fba, 0x0a637dc5a2c898a6, 0x113f9804bef90dae, 0x1b710b35131c471b,
+            0x28db77f523047d84, 0x32caab7b40c72493, 0x3c9ebe0a15c9bebc, 0x431d67c49c100d4c,
+            0x4cc5d4becb3e42b6, 0x597f299cfc657e2a, 0x5fcb6fab3ad6faec, 0x6c44198c4a475817]
+
+        var accumulated = self
+        let lengthInBits = accumulated.count * 8
+        let lengthBytes = lengthInBits.bytes(count: blockSize / 8)
+        
+        // Step 1: Append padding.
+        let msgLength = accumulated.count
+        // Append one bit(`UInt8` with one bit) to the message.
+        accumulated.append(0x80)
+        // Append `0` bits until the length of `accumulated` in bits is 448 (mod 512).
+        let max = blockSize * 7 / 8
+        accumulated += [UInt8](
+            repeating: 0,
+            count: msgLength % blockSize < max ?
+                max - 1 - (msgLength % blockSize) :
+                blockSize + max - 1 - (msgLength % blockSize))
+
+        // Step 2: Append the message length as a 64-bit representation of `lengthInBits`.
+        accumulated += lengthBytes
+
+        // Step 3: Process the array bytes.
+        var accumulatedHash: [UInt64] = [
+            0x6a09e667f3bcc908, 0xbb67ae8584caa73b, 0x3c6ef372fe94f82b, 0xa54ff53a5f1d36f1, 
+            0x510e527fade682d1, 0x9b05688c2b3e6c1f, 0x1f83d9abfb41bd6b, 0x5be0cd19137e2179]
+        var index = 0
+        while index < accumulated.count {
+            let chunk = accumulated[index..<(index + blockSize)]
+            index += blockSize
+            
+            // Break chunk into sixteen 64-bit words M[j], 0 ≤ j ≤ 15, in big-endian format.
+            // Extend the sixteen 64-bit words into eighty 64-bit words:
+            let M = UnsafeMutablePointer<UInt64>.allocate(capacity: k.count)
```

Comment:
I'd suggest using an array here: `let M = Array(repeating: 0, count: k.count)`, which is safer and will be promoted to a stack allocation in the optimizer.